aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -19,20 +19,82 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 #include <c10/cuda/CUDAStream.h>
 
+// TODO(NS): Investigate why FP8 conversion intrisncs end up being slower
+#if 0
+#include <cuda_fp8.h>
+#endif
+
 namespace at::native {
 
 void neg_kernel_cuda(TensorIteratorBase &iter);
 void conj_kernel_cuda(TensorIteratorBase &iter);
 
+void float8_copy_kernel_cuda(TensorIteratorBase &iter) {
+  ScalarType dtype = iter.dtype(0);
+  ScalarType other_dtype = iter.dtype(1);
+  if (dtype == kFloat8_e4m3fn) {
+    if (other_dtype == kFloat) {
+       gpu_kernel_nocast(iter, [] GPU_LAMBDA(float value) {
+           return Float8_e4m3fn(value);
+       });
+    } else if (other_dtype == kHalf) {
+       gpu_kernel_nocast(iter, [] GPU_LAMBDA(Half value) {
+           return Float8_e4m3fn(value);
+       });
+    } else if (other_dtype == kBFloat16) {
+       gpu_kernel_nocast(iter, [] GPU_LAMBDA(BFloat16 value) {
+           return Float8_e4m3fn(value);
+       });
+    } else {
+      gpu_kernel(iter, [] GPU_LAMBDA(Float8_e4m3fn x) { return x; });
+    }
+  } else if (dtype == kFloat8_e5m2) {
+    if (other_dtype == kFloat) {
+       gpu_kernel_nocast(iter, [] GPU_LAMBDA(float value) {
+#if 0
+           const auto x =  __nv_cvt_float_to_fp8(value, __NV_NOSAT, __NV_E5M2);
+           return Float8_e5m2(x, Float8_e5m2::from_bits());
+#else
+           return Float8_e5m2(value);
+#endif
+       });
+    } else if (other_dtype == kHalf) {
+       gpu_kernel_nocast(iter, [] GPU_LAMBDA(Half value) {
+#if 0
+           const auto x =  __nv_cvt_halfraw_to_fp8(static_cast<__half>(value), __NV_NOSAT, __NV_E5M2);
+           return Float8_e5m2(x, Float8_e5m2::from_bits());
+#else
+           return Float8_e5m2(value);
+#endif
+       });
+    } else if (dtype == kFloat8_e5m2 && other_dtype == kBFloat16) {
+       gpu_kernel_nocast(iter, [] GPU_LAMBDA(BFloat16 value) {
+#if 0
```

Comment:
Nit: should we create the macro now and just set it to 0

Reply:
Sorry, those supposed to be guarded by the macro, let me replace this one..